tests/foreman/cli/test_activationkey.py

```diff
@@ -264,28 +234,31 @@ def test_positive_create_activation_key_4(self, test_data):
         @Assert: Activation key is created
 
         """
-        try:
-            org_obj = make_org(cached=True)
-            con_view = make_content_view({
-                u'organization-id': org_obj['id'],
-                u'name': test_data['content-view']
-            })
-            new_ackey = self._make_activation_key({
-                u'name': test_data['name'],
-                u'content-view': con_view['name'],
-                u'environment': self.library['name'],
-                u'organization-id': org_obj['id']
-            })
-        except CLIFactoryError as err:
-            self.fail(err)
-
-        # Name should match passed data
-        self.assertEqual(new_ackey['name'], test_data['name'])
-        # ContentView should match passed data
-        self.assertEqual(new_ackey['content-view'], test_data['content-view'])
+        for name in valid_data_list():
+            with self.subTest(name):
+                try:
+                    org_obj = make_org(cached=True)
```

Comment:
I'd recommend moving the org creation outside of the loop and possibly without `cached=True` as the same org would be used for the permutations inside the `subTest` loop